timeseries/src/autogluon/timeseries/models/tabular/tabular_model.py

```diff
@@ -0,0 +1,198 @@
+
+    - lag features (observed time series values) based on ``freq`` of the data
+    - time features (e.g., day of the week) based on the timestamp of the measurement
+    """
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        name = name or re.sub(r"Model$", "", self.__class__.__name__)  # TODO: look name up from presets
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+        self._lag_indices: List[int] = None
+        self._time_features: List[TimeFeature] = None
+        self._available_features: pd.Index = None
+        self._drop_median_prediction = False
+
+        if 0.5 not in self.quantile_levels:
+            self._drop_median_prediction = True
+            self.quantile_levels = sorted(self.quantile_levels + [0.5])
+
+        self.tabular_predictor = TabularPredictor(
+            label=self.target,
+            problem_type=ag.constants.QUANTILE,
+            quantile_levels=self.quantile_levels,
+        )
+
+    def _get_features_dataframe(
+        self,
+        data: TimeSeriesDataFrame,
+        last_k_values: Optional[int] = None,
+    ) -> pd.DataFrame:
+        """Generate a feature matrix used by TabularPredictor.
+
+        Parameters
+        ----------
+        data : TimeSeriesDataFrame
+            Dataframe containing features derived from time index & past time series values, as well as the target.
+        last_k_values: int, optional
+            If provided, features will be generated only for the last `last_k_values` timesteps of each time series.
+        """
+        if last_k_values is None:
+            selected_slice = slice(None, None)
+        else:
+            selected_slice = slice(-last_k_values, None)
+
+        # TODO Parrallelize the code
+        dataframe_per_item = []
+        for item_id in data.item_ids:
+            series = data.loc[item_id][self.target]
+            time_feature_columns = {
+                feature.__class__.__name__: feature(series.index[selected_slice]) for feature in self._time_features
+            }
+            lag_columns = {f"lag_{idx}": series.shift(idx).values.ravel()[selected_slice] for idx in self._lag_indices}
+            columns = {**time_feature_columns, **lag_columns, "target": series.values.ravel()[selected_slice]}
+            df = pd.DataFrame(columns, index=series.index[selected_slice])
+            dataframe_per_item.append(df.iloc[selected_slice])
+
+        return pd.concat(dataframe_per_item, ignore_index=True)
+
+    def _generate_features_from_freq(self, freq: str):
```

Comment:
nit: this is not really generating the features right? A somewhat more cohesive way is to fold this under `_initialize` (see `AbstractTimeSeriesModel`)

```
def _initialize(..):
    super()._initialize()
    self._lag_indices = ..
```

Reply:
This depends on `train_data.freq` that we only obtain during `_fit`, so doing this inside `_initialize` is too early. I will move this code inside `_fit`.